src/test/java/org/jenkinsci/plugins/workflow/properties/PipelineTriggersJobPropertyTest.java

```diff
@@ -0,0 +1,182 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class PipelineTriggersJobPropertyTest {
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    /**
+     * Needed to ensure that we get a fresh {@code startsAndStops} with each test run. Has to be *after* rather than
+     * *before* to avoid weird ordering issues with {@code @LocalData}.
+     */
+    @After
+    public void resetStartsAndStops() {
+        MockTrigger.startsAndStops = new ArrayList<>();
+    }
+
+    @Test
+    public void loadCallsStartFalse() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        p = (WorkflowJob) Items.load(p.getParent(), p.getRootDir());
+        t = (MockTrigger) p.getTriggers().get(t.getDescriptor());
+        assertNotNull(t);
+        // The  first "null, false" is due to the p.addTrigger(t) call and won't apply in the real world.
+        assertEquals("[null, false, null, false]", MockTrigger.startsAndStops.toString());
+        Boolean currentStatus = t.currentStatus();
+        assertNotNull(currentStatus);
+        assertFalse(currentStatus);
+    }
+
+    @Test
+    public void submitCallsStartTrue() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        p = (WorkflowJob) r.configRoundtrip((Item) p);
+        t = (MockTrigger) p.getTriggers().get(t.getDescriptor());
+        assertNotNull(t);
+        // The  first "null, false" is due to the p.addTrigger(t) call and won't apply in the real world.
+        assertEquals("[null, false, null, true]", MockTrigger.startsAndStops.toString());
+        Boolean currentStatus = t.currentStatus();
+        assertNotNull(currentStatus);
+        assertTrue(currentStatus);
+    }
+
+    @Test
+    public void previousTriggerStopped() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        MockTrigger t2 = new MockTrigger();
+        p.addTrigger(t2);
+        p.save();
+
+        assertFalse(t.isStarted);
+        assertTrue(t2.isStarted);
+    }
+
+    @LocalData
+    @Test
+    public void triggerMigration() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+        assertNotNull(p);
+
+        PipelineTriggersJobProperty triggerProp = p.getProperty(PipelineTriggersJobProperty.class);
+        assertNotNull(triggerProp);
+
+        assertEquals(2, triggerProp.getTriggers().size());
+        assertEquals(2, p.getTriggers().size());
```

Comment:
Seems like it would be simpler to just assert that these two lists are equal, then check each element once?